### PR TITLE
Add some functions for starting Pegasus in GraphScope

### PR DESCRIPTION
**Pull request by BingqingLyu**
## Related issue number

Fixes

### CODE REVIEW

File: research/engine/pegasus/pegasus/src/lib.rs [hunk 1]
```diff
@@ -324,6 +327,7 @@ fn allocate_local_worker(conf: &Arc<JobConf>) -> Result<Option<WorkerIdIter>, Bu
 }
 
 use std::sync::Once;
+use std::net::SocketAddr;
```
Comment (longbinlai) [inline research/engine/pegasus/pegasus/src/lib.rs:330]: Move these `use` to the head of the file

File: research/engine/pegasus/pegasus/src/lib.rs [hunk 2]
```diff
@@ -183,13 +183,16 @@ pub fn startup_with<D: ServerDetect + 'static>(conf: Configuration, detect: D) -
...
         let conn_conf = net_conf.get_connection_param();
         let addr = pegasus_network::start_up(server_id, conn_conf, addr, detect)?;
         info!("server {} start on {:?}", server_id, addr);
-    }
-    Ok(())
+        Some(addr)
+    } else {
+        None
+    };
+    Ok(res)
```
Comment (longbinlai) [inline research/engine/pegasus/pegasus/src/lib.rs:195]: Directly return `Ok(if let Some(...) {} else { ... })`, instead of `let res = if let ...; Ok(res)`

**Review by longbinlai — COMMENTED** [review]
Small fix.

File: research/engine/pegasus/pegasus/src/lib.rs (hunk 1 above)
Comment (BingqingLyu) [reply research/engine/pegasus/pegasus/src/lib.rs]: Done

File: research/engine/pegasus/pegasus/src/lib.rs (hunk 2 above)
Comment (BingqingLyu) [reply research/engine/pegasus/pegasus/src/lib.rs]: Done

File: research/engine/pegasus/network/src/manager.rs [hunk 3]
```diff
@@ -58,7 +58,36 @@ impl ServerManager {
...
+}
+
+impl SimpleServerDetector {
+    pub fn new() -> Self {
+        SimpleServerDetector { peers_mutex: Mutex::new(vec![]) }
+    }
+
+    pub fn update_peer_view<Iter: Iterator<Item = (u64, SocketAddr)>>(&self, peer_view: Iter) {
+        let new_peers = peer_view.map(|(id, addr)| Server { id, addr }).collect::<Vec<Server>>();
+        let mut peers = self.peers_mutex.lock().expect("unexpected error locking when update peer view");
```
Comment (longbinlai) [inline research/engine/pegasus/network/src/manager.rs:77]: Will this `expect()` cause any issue? I mean if it is used while running a service, then it should not be down due to such an error; if it is just for initiating a service, then it is fine.

File: research/engine/pegasus/network/src/manager.rs [hunk 4]
```diff
@@ -58,7 +58,36 @@ impl ServerManager {
...
+        *peers = new_peers;
+    }
+}
+
+impl ServerDetect for SimpleServerDetector {
+    fn fetch(&self) -> Vec<Server> {
+        let peers = self.peers_mutex.lock().expect("unexpected error locking when fetch servers");
```
Comment (longbinlai) [inline research/engine/pegasus/network/src/manager.rs:84]: and this `expect()`.

File: research/engine/pegasus/network/src/manager.rs (hunk 3 above)
Comment (BingqingLyu) [reply research/engine/pegasus/network/src/manager.rs:77]: This is just for initiating the service (used when starting service with v2 store).

**Pull request merged**